cdap-proto/src/main/java/io/cdap/cdap/proto/ApplicationUpdateDetail.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.proto;
+
+import io.cdap.cdap.api.common.HttpErrorStatusProvider;
+import io.cdap.cdap.proto.id.ApplicationId;
+import javax.annotation.Nullable;
+
+/**
+ * Represents an application update result of an {@link ApplicationDetail}.
+ */
+public class ApplicationUpdateDetail {
+
+  private final int statusCode;
+  private final String error;
+  private final String updateDetails;
+  private final ApplicationId applicationId;
```

Comment:
this should match what is given as the input except have additional fields for status code and message. The javadoc for the REST API seems to indicate the input is an 'appId' variable of type string.

Reply:
looking at some of the other batch endpoints, they return lists of objects like BatchProgramResult. There, appId is a flat field. It also only has an 'error' string and nothing like 'updateDetails'. If the operation is successful, there is no message.

Reply:
Done.